src/main/java/com/mongodb/jdbc/MongoStatement.java

```diff
@@ -187,45 +186,181 @@ public boolean execute(String sql) throws SQLException {
         return resultSet != null;
     }
 
+    private ResultSet executeAtlasDataFederationQuery(String sql) throws SQLException {
+        BsonDocument getSchemaCmd = constructSQLGetResultSchemaDocument(sql);
+
+        BsonDocument sqlStage = constructQueryDocument(sql);
+        MongoIterable<BsonDocument> iterable =
+                currentDB
+                        .aggregate(Collections.singletonList(sqlStage), BsonDocument.class)
+                        .maxTime(maxQuerySec, TimeUnit.SECONDS);
+
+        if (fetchSize != 0) {
+            iterable = iterable.batchSize(fetchSize);
+        }
+
+        MongoCursor<BsonDocument> cursor = iterable.cursor();
+        MongoJsonSchemaResult schemaResult =
+                currentDB.runCommand(getSchemaCmd, MongoJsonSchemaResult.class);
+        MongoJsonSchema schema = schemaResult.schema.mongoJsonSchema;
+        List<List<String>> selectOrder = schemaResult.selectOrder;
+
+        resultSet =
+                new MongoResultSet(
+                        this,
+                        cursor,
+                        schema,
+                        selectOrder,
+                        conn.getExtJsonMode(),
+                        conn.getUuidRepresentation());
+
+        return resultSet;
+    }
+
+    private ResultSet executeDirectClusterQuery(String sql) throws SQLException {
+        Map<String, List<String>> namespaces =
+                conn.getRunCmd().getNamespaces(currentDB.getName(), sql);
+
+        if (namespaces.size() > 1) {
+            throw new SQLException(
+                    "Multiple databases are involved in the query. Only one database can be processed.");
+        }
+
+        String dbName = currentDB.getName();
+        List<String> collections = namespaces.get(dbName);
+        if (collections == null) {
+            throw new SQLException("No collections found for the current database: " + dbName);
+        }
+
+        Document catalogDoc = buildCatalogDocument(dbName, collections);
+        // Setting excludeNamespaces and relaxSchemaChecking to default values false. These options are not currently
+        // handled in the JDBC driver
+        Document translateResponseDoc =
+                conn.getRunCmd().translate(sql, dbName, false, false, catalogDoc);
```

Comment:
I set the excludeNamespaces and relaxSchemaChecking to false here because we don't have options in the driver to allow these to be configured.  Also when I initially had excludeNamespaces set to true it would fail on parsing the select order because it was only getting the `field` instead of `datasource`, `field`.  So there seems to be more work there.  Not sure if these would even be useful to be set through the JDBC driver.